File: src/matlab/spacedyn_v2r0/eul2dc.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% The SpaceDyn, a MATLAB toolbox for Space and Mobile Robots.
% (C)1998 The Space Robotics Lab. directed by Kazuya Yoshida,
% Tohoku University, Japan.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%   version 1 // Oct 4, 1999, Last modification by K.Yoshida
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%
% 	EUL2DC	3-1-3 Euler angles to transformation matrix
%
%		EUL2DC(Z1,X2,Z3) returns a Direction Cosine matrix 
%		for the specified Euler angles.  
%		These correspond to rotations about the
%		Z, X, Z axes respectively.
%
%		See also DC2EUL, RPY2DC.
%

function C = eul2dc( phi , theta , psi )


if length( phi ) == 3
   C = cz( phi(3) ) * cx( phi(2) ) * cz( phi(1) );
   
else
   C = cz( psi ) * cx( theta ) * cz( phi );
   
end

%%% EOF

File: eul2dc.py
import numpy as np
from Cxyz import cxyz


def eul2dc(phi, theta, psi):
    C = cxyz(psi, 0, 0, 1) @ cxyz(theta, 1, 0, 0) @ cxyz(phi, 0, 0, 1)

    # if np.size(phi == 3):
    #     C = cxyz(phi[2], 0, 0, 1) @ cxyz(phi[1], 1, 0, 0) @ cxyz(phi[0], 0, 0, 1)
    # else:
    #     C = cxyz(psi, 0, 0, 1) @ cxyz(theta, 1, 0, 0) @ cxyz(phi, 0, 0, 1)

    return C
